src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -77,9 +77,9 @@ protected function completeOrder(OrderInterface $order)
 
         $this->applyTransition(OrderCheckoutTransitions::TRANSITION_COMPLETE, $order);
         $this->get('sm.factory')->get($order, OrderTransitions::GRAPH)->apply(OrderTransitions::SYLIUS_CREATE, true);
-        if ($order->getCurrency() !== $currencyProvider->getBaseCurrency()) {
+        if ($order->getCurrencyCode() !== $currencyProvider->getBaseCurrency()) {
```

Comment:
`$currencyProvider->getBaseCurrency()` returns `Currency` object, so using `!==` will always return true. IMO we can remove this `if` and left the logic to always save the exchange rate on `Order`.